compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -1827,6 +1834,12 @@ class Typer extends Namer
               case mdef1: DefDef if Inliner.hasBodyToInline(mdef1.symbol) =>
                 buf ++= inlineExpansion(mdef1)
               case mdef1 =>
+                import untpd.modsDeco
+                mdef match {
+                  case mdef: untpd.TypeDef if mdef.mods.hasMod[untpd.Mod.Enum] =>
+                    enumContexts(mdef1.symbol) = ctx
+                  case _ =>
+                }
```

Comment:
`ctx` is not actually the context (which contains class members) used in checking member definitions, thus the following code fails the check:

```Scala
enum Color(val x: Int) {
  case Green  extends Color(3)
  case Red    extends Color(2)
  case Violet extends Color(Green.x + Red.x)        // error
}
```

Is this intentional?

Reply:
Yes, I think we need a refinement here. This accessibility checking is quite a can of worms.